applications/luci-app-upnp/htdocs/luci-static/resources/view/upnp/upnp.js

```diff
@@ -107,67 +107,73 @@ return view.extend({
+			_('Allow adding port maps only to requesting IP addresses and deny third-party PCP option')).default = '1'
 
-		s.taboption('general', form.Value, 'upload', _('Upload speed'),
-			_('Value in KByte/s, informational only')).rmempty = true
+		o = s.taboption('advanced', form.Value, 'notify_interval', _('Notify interval'), _('A 900s interval will result in %s notifications with the minimum max-age of 1800s', 'A 900s interval will result in %s (%s = SSDP) notifications with the minimum max-age of 1800s').format('<abbr title="Simple Service Discovery Protocol">SSDP</abbr>'))
+		o.depends('enable_upnp', '1')
+		o.datatype    = 'uinteger'
+		o.placeholder = 900
 
-		o = s.taboption('general', form.Value, 'port', _('Port'))
+		o = s.taboption('advanced', form.Value, 'port', _('SOAP/HTTP port'))
+		o.depends('enable_upnp', '1')
 		o.datatype = 'port'
 		o.default  = 5000
 
-		s.taboption('advanced', form.Flag, 'system_uptime', _('Report system instead of service uptime')).default = '1'
-
-		s.taboption('advanced', form.Value, 'uuid', _('Device UUID'))
-		s.taboption('advanced', form.Value, 'serial_number', _('Announced serial number'))
-		s.taboption('advanced', form.Value, 'model_number', _('Announced model number'))
+		o = s.taboption('advanced', form.Value, 'presentation_url', _('Presentation URL'))
+		o.depends('enable_upnp', '1')
+		o.placeholder = 'http://192.168.1.1/'
 
-		o = s.taboption('advanced', form.Value, 'notify_interval', _('Notify interval'))
-		o.datatype    = 'uinteger'
-		o.placeholder = 30
+		o = s.taboption('advanced', form.Value, 'uuid', _('Device UUID'))
+		o.depends('enable_upnp', '1')
+		o = s.taboption('advanced', form.Value, 'model_number', _('Announced model number'))
+		o.depends('enable_upnp', '1')
+		o = s.taboption('advanced', form.Value, 'serial_number', _('Announced serial number'))
+		o.depends('enable_upnp', '1')
 
-		o = s.taboption('advanced', form.Value, 'clean_ruleset_threshold', _('Clean rules threshold'))
-		o.datatype    = 'uinteger'
-		o.placeholder = 20
+		o = s.taboption('advanced', form.Flag, 'system_uptime', _('Report system instead of service uptime'))
+		o.depends('enable_upnp', '1')
+		o.default = '1'
 
-		o = s.taboption('advanced', form.Value, 'clean_ruleset_interval', _('Clean rules interval'))
-		o.datatype    = 'uinteger'
-		o.placeholder = 600
-
-		o = s.taboption('advanced', form.Value, 'presentation_url', _('Presentation URL'))
-		o.placeholder = 'http://192.168.1.1/'
+		s.taboption('advanced', form.Flag, 'log_output', _('Enable additional logging'),
+			_('Puts extra debugging information into the system log'))
 
 		o = s.taboption('advanced', form.Value, 'upnp_lease_file', _('Service lease file'))
 		o.placeholder = '/var/run/miniupnpd.leases'
 
-		s.taboption('advanced', form.Flag, 'use_stun', _('Use STUN'))
+		s.taboption('advanced', form.Flag, 'use_stun', _('Use STUN'), _('Useful to detect the public IPv4 address for unrestricted full-cone, aka one-to-one, NATs'))
```

Comment:
This is OK. Although an improvement would be to use the abbr as used elsewhere for the STUN acronym, since it is infrequently encountered by home-users. Alt. Point a URI to either an English wikipedia entry, or an appropriate RFC.